server/plugin.go

```diff
@@ -0,0 +1,379 @@
+	"github.com/mattermost/mattermost-server/v5/plugin"
+	"github.com/pkg/errors"
+)
+
+const jitsiNameSchemaAsk = "ask"
+const jitsiNameSchemaEnglish = "english-titlecase"
+const jitsiNameSchemaUUID = "uuid"
+const jitsiNameSchemaMattermost = "mattermost"
+const configChangeEvent = "config_update"
+
+type UserConfig struct {
+	Embedded     bool   `json:"embedded"`
+	NamingScheme string `json:"naming_scheme"`
+}
+
+type Plugin struct {
+	plugin.MattermostPlugin
+
+	// configurationLock synchronizes access to the configuration.
+	configurationLock sync.RWMutex
+
+	// configuration is the active plugin configuration. Consult getConfiguration and
+	// setConfiguration for usage.
+	configuration *configuration
+}
+
+func (p *Plugin) OnActivate() error {
+	config := p.getConfiguration()
+	if err := config.IsValid(); err != nil {
+		return err
+	}
+
+	if err := p.API.RegisterCommand(createJitsiCommand()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type User struct {
+	Avatar string `json:"avatar"`
+	Name   string `json:"name"`
+	Email  string `json:"email"`
+	ID     string `json:"id"`
+}
+
+type Context struct {
+	User  User   `json:"user"`
+	Group string `json:"group"`
+}
+
+type EnrichMeetingJwtRequest struct {
+	Jwt string `json:"jwt"`
+}
+
+// Claims extents cristalhq/jwt standard claims to add jitsi-web-token specific fields
+type Claims struct {
+	jwt.StandardClaims
+	Context Context `json:"context"`
+	Room    string  `json:"room,omitempty"`
+}
+
+func verifyJwt(secret string, jwtToken string) (*Claims, error) {
+	verifier, err := jwt.NewVerifierHS(jwt.HS256, []byte(secret))
+	if err != nil {
+		log.Printf("Error generating new HS256 signer: %v", err)
+		return nil, err
+	}
+
+	newToken, err := jwt.ParseAndVerifyString(jwtToken, verifier)
+	if err != nil {
+		log.Printf("Error parsing or verifiying jwt: %v", err)
+		return nil, err
+	}
+
+	var claims Claims
+	if err = json.Unmarshal(newToken.RawClaims(), &claims); err != nil {
+		log.Printf("Error unmarshalling claims from jwt: %v", err)
+		return nil, err
+	}
+	return &claims, nil
+}
+
+func signClaims(secret string, claims *Claims) (string, error) {
+	signer, err2 := jwt.NewSignerHS(jwt.HS256, []byte(secret))
+	if err2 != nil {
+		log.Printf("Error generating new HS256 signer: %v", err2)
+		return "", errors.New("internal error")
+	}
+	builder := jwt.NewBuilder(signer)
+	token, err := builder.Build(claims)
+	if err != nil {
+		return "", err
+	}
+	return string(token.Raw()), nil
+}
+
+func (p *Plugin) deleteEphemeralPost(userID, postID string) {
+	p.API.DeleteEphemeralPost(userID, postID)
+}
+
+func (p *Plugin) updateJwtUserInfo(jwtToken string, user *model.User) (string, error) {
+	secret := p.getConfiguration().JitsiAppSecret
+
+	claims, err := verifyJwt(secret, jwtToken)
+	if err != nil {
+		return "", err
+	}
+
+	config := p.API.GetConfig()
+	if config.PrivacySettings.ShowFullName == nil || !*config.PrivacySettings.ShowFullName {
+		user.FirstName = ""
```

Comment:
Probably not a real problem, but I do not like these statements modifying a parameter passed by reference. Probably in its current context makes no harm, but may provoke unexpected errors on future iterations of the plugin.

Reply:
fair enough

Reply:
Done, look at the last added commit.